cmd/federation-controller/main.go

```diff
@@ -168,16 +168,26 @@ func main() {
 		kube.NewWorkloadEntryReconciler(istioClient, istioConfigFactory),
 		kube.NewPeerAuthResourceReconciler(istioClient, namespace),
 	}
+	if cfg.MeshPeers.Remote.IngressType == config.OpenShiftRouter {
+		reconcilers = append(reconcilers, kube.NewDestinationRuleReconciler(istioClient, istioConfigFactory))
+	}
 	if cfg.MeshPeers.Local.IngressType == config.OpenShiftRouter {
 		routeClient, err := routev1client.NewForConfig(kubeConfig)
 		if err != nil {
 			log.Fatalf("failed to create Route client: %v", err)
 		}
 
-		reconcilers = append(reconcilers, kube.NewDestinationRuleReconciler(istioClient, istioConfigFactory))
 		reconcilers = append(reconcilers, kube.NewEnvoyFilterReconciler(istioClient, istioConfigFactory))
 		reconcilers = append(reconcilers, kube.NewRouteReconciler(routeClient, openshift.NewConfigFactory(*cfg, serviceLister)))
+	}
 
+	rm := kube.NewReconcilerManager(meshConfigPushRequests, reconcilers...)
+	if err := rm.ReconcileAll(ctx); err != nil {
+		log.Fatalf("initial Istio resource reconciliation failed: %v", err)
+	}
+	go rm.Start(ctx)
+
+	if !networking.IsIP(remote.Addresses[0]) {
```

Comment:
In the future we could start the resolver only if a locally existing service was imported (will create an issue).